### PR TITLE
[FLINK-32234][table] Support execute truncate table statement

### PR DESCRIPTION
**Pull request by luoyuxia**
## What is the purpose of the change
To make Flink support to execute truncate table statement

## Brief change log
  -  The first commit convert `SQLTruncateTable` to `TruncateTableOperation`
  -  The second commit allow to execute `TruncateTableOperation`
  -  The third commit add test in sql-client for truncate table statement

## Verifying this change
UT & IT

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes / **no**)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (yes / **no**)
  - The serializers: (yes / **no** / don't know)
  - The runtime per-record code paths (performance sensitive): (yes / **no** / don't know)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: (yes / **no** / don't know)
  - The S3 file system connector: (yes / **no** / don't know)

## Documentation

  - Does this pull request introduce a new feature? (yes / **no**)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / **not documented**)

### CODE REVIEW

File: flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/operations/TruncateTableOperation.java [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+
+    public TruncateTableOperation(ObjectIdentifier tableIdentifier) {
+        this.tableIdentifier = tableIdentifier;
+    }
+
+    @Override
+    public TableResultInternal execute(Context ctx) {
+        if (ctx.isStreamingMode()) {
+            throw new TableException(
+                    "TRUNCATE TABLE statement is not supported in streaming mode" + ".");
```
Comment (Tartarus0zm) [inline flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/operations/TruncateTableOperation.java:50]:
```suggestion
                    "TRUNCATE TABLE statement is not supported in streaming mode.");
```

**Review by Tartarus0zm — COMMENTED** [review]
@luoyuxia  Thanks for this pr. Just a small comment.

**Comment by luoyuxia** [comment]
@Tartarus0zm Thanks for your review. I have addressed your comments. PTAL.

**Comment by luoyuxia** [comment]
@flinkbot run azure

**Review by Tartarus0zm — APPROVED** [review]
LGTM

**Comment by luoyuxia** [comment]
@Tartarus0zm Thanks for reviewing...

**Pull request closed without merging**